api/proto/config/edgeview.proto

```diff
@@ -0,0 +1,56 @@
+// Copyright(c) 2021 Zededa, Inc.
+// All rights reserved.
+
+syntax = "proto3";
+
+package org.lfedge.eve.config;
+option go_package  = "github.com/lf-edge/eve/api/go/config";
+option java_package = "org.lfedge.eve.config";
+
+message EdgeViewConfig {
+    // JWT token for signed info, it contains the dispatcher
+    // endpoint IP:Port, device UUID and expiration time
+    string  token = 1;
+    // dispatcher certificate(s) if it's not well-known CA signed
+    repeated bytes disp_cert_pem = 2;
+    // policy for device access through edge-view
+    DevDebugAccessPolicy dev_policy = 3;
+    // policy access for apps through edge-view
+    AppDebugAccessPolicy app_policy = 4;
+}
+
+// Dev debug policy applicable to edge-view
+message DevDebugAccessPolicy {
+    // device side of edge-view access is allowed or not
+    bool allow_dev = 1;
+    // allow edge-view ssh into device or not, ssh key still need to be
+    // installed separately even the edge-view allows it
+    bool ssh = 2;
+    // allow edge-view shell/* cmd or not
+    bool allow_shell = 3;
+    // allow to issue edge-view shell reboot device or not, provided allow_shell is true
+    bool shell_reboot = 4;
+    // allow edge-view to launch VNC into any application console on device or not; the
+    // app still need to enable VNC access even it is allowed here
+    bool app_console = 5;
+}
+
+// App debug policy applicable to edge-view
+message AppDebugAccessPolicy {
+    // app side of edge-view access is allowed or not
+    bool allow_app = 1;
+    // ssh through port 22 into app is allowed or not
+    bool ssh = 2;
+    // the https proxy is allowed or not
+    bool proxy = 3;
+    // only the defined set of tcp ports allowed
+    // if none is configured, all ports are allowed except for explicit disallow_ports
+    repeated int32 only_tcp_ports = 4;
+    // the ports listed are not allowed
+    repeated int32 disallow_tcp_ports = 5;
+    // only the listed IP-Addresses or Domain Names are allowed
+    // if none is configured, all addresses are allowed except for explicit disallow_addrs
+    repeated string only_addrs = 6;
+    // the IP-Addresses or Domain Names are not allowed
+    repeated string disallow_addrs = 7;
```

Comment:
These are the remote IP address which are allowed to connect, correct?

But what does that mean? My laptop has IP address 192.168.1.11, but it has a different public IP address, and the dispatcher has a third IP address.

If we want to limit different remote users and devices to access certain edge-view features, then instead of depending on their IP addresses it seems simpler and more flexible to specify a list of features when the JWT is created, and potentially allow creating multiple JWTs with different features and giving them to different users which need to do debugging.

Reply:
No, the intention of this 'addrs' is for destination ip, is either for app ip addresses, or the 'localhost' (which is dom0 side) like of addresses. they can also be some servers on the same network as the eve-device, say a local-profile server. those are not to limit the user or source.

Reply:
Ah, that wasn't clear from the API definition/comments.
I haven't thought of that general concept but shouldn't we also specify the network instance to which the access is granted? If the future we might have the same private IP address on different network instances (I think this works today for switch network instances.)

In addition, for app instances deployed on the device the user might know the app instance name or UUID, but the IP address is assigned as it is deployed hence not known in advance. Thus there might be two variants: 
1. access app instance X (over the default network instance or a specified network instance)
2. access IP address Y (whether on the device or outside) over a specified network instance

Reply:
Sure. for 1) above, 'edge-view' container can get the app info first and verify it with the tcp tunnel ip-destination ip address